best-practices/go-ci.md

```diff
@@ -0,0 +1,54 @@
+## Best practices for Continuous Integration with Go
+
+First, use [Modules](https://golang.org/ref/mod) even if you don't have any
+external dependencies. Beyond ensuring consistency, a module allows using new Go
+language features. The instructions below assume you are setting up CI for a
+single Go module.
+
+It is strongly recommended that the following checks are used:
+
+* `go test ./...` to run all tests
+* `go test -race ./...` to catch any data races when running the tests
+* `test -z $(gofmt -l .)` to ensure all Go files are well formatted
+* `go vet ./...` to ensure there are no vet warnings
+
+The following checks are also recommended, though they might not help in all
+situations:
+
+* `go mod tidy && git diff --exit-code` to ensure `go.mod` and `go.sum` are clean
+* `go mod verify` to ensure that `go.sum` agrees with what's in the module cache
```

Comment:
This matters if you downloaded the modules at some point in the past, and since then the cached downloads have been tampered with. Because your filesystem is faulty or because someone tampered with them while your machine was unattended, for example. Go will only checksum dependencies when downloading them; once cached, it will assume that the local filesystem is trustworthy. Otherwise, having to re-validate deps at every build would be quite expensive.

You're right that, in most cases, this will not matter for CI. Most people don't even set up caching for their CI. That is why this suggestion is left optional. But it's still a valuable one if the CI uses a cache, and one wants to make sure that the cache hasn't been tampered with, or hasn't been broken unintentionally. If one was using CI/CD, that kind of problem could result in deploying faulty binaries.

Reply:
Thanks for the explanation!